chemprop/schedulers.py

```diff
@@ -1,11 +1,16 @@
-import numpy as np
-from numpy.typing import ArrayLike
 from torch.optim import Optimizer
-from torch.optim.lr_scheduler import LRScheduler
-
-
-class NoamLR(LRScheduler):
-    r"""A Noam learning rate scheduler schedules the learning rate with a piecewise linear followed
+from torch.optim.lr_scheduler import LambdaLR
+
+
+def get_NoamLR_sched(
```

Comment:
I am not sure if this would be a better name
```suggestion
def get_linear_warmup_exponential_decay_sched(
```

Reply:
```suggestion
def build_NoamLike_LRSched(
```
I prefer `build` as the verb here. And we should update the docstring and name to reflect that this scheduler is "Noam-like" rather than actually a Noam scheduler.

Reply:
Looks good! I have adopted it.